File: connection-examples/nodejs/connection.js
const { Client } = require('pg');

const client = new Client({
    user: "MATERIALIZE_USERNAME",
    password: "MATERIALIZE_PASSWORD",
    host: "MATERIALIZE_HOST",
    port: 6875,
    database: 'materialize',
    ssl: true
});

/*
    Alternatively, you can use the following syntax:
    const client = new Client('postgres://materialize@localhost:6875/materialize');
*/

async function main() {
    await client.connect();
    /* Work with Materialize */
}

main();

File: connection-examples/python/connection.py
#!/usr/bin/env python3

import psycopg2
import sys

dsn = "user=MATERIALIZE_USERNAME password=MATERIALIZE_PASSWORD host=MATERIALIZE_HOST port=6875 dbname=materialize sslmode=require"
# Alternative syntax:
# dsn = "postgresql://MATERIALIZE_USERNAME:MATERIALIZE_PASSWORD@MATERIALIZE_HOST:6875/materialize?sslmode=require"
conn = psycopg2.connect(dsn)
